### ISSUE TITLE
Why logsoftmax in the expert's output?

### ISSUE THREAD

**Issue by sofiapvcp** [davidmrau/mixture-of-experts#13]
Hi, thanks for the rep! 

Could you kindly clarify why do you use `nn.LogSoftmax( )` in the last layer of the MLP experts? Thanks!

I'm guessing the reason is because you are using NLL loss in one of the examples. However, in the CIFAR example crossentropyloss is used.

**Comment by davidmrau** [comment]
Hi @sofiapvcp!
You are absolutely right, this is just legacy code from one example and should be changed. If you want you could improve the code by making it a bit more modular :)

**Comment by xpai** [comment]
Hi, I am wondering why the expert combine function should keep in the log space and use an exp-then-log transformation. https://github.com/davidmrau/mixture-of-experts/blob/master/moe.py#L97
Is it the same reason as the above?

**Comment by davidmrau** [comment]
@xpai I believe the reason for transforming between log and exp-then-log space was to conveniently be able to use the index_add function. As far as I know index_multiply doesn't exist...

**Comment by davidmrau** [comment]
@xpai please see https://github.com/davidmrau/mixture-of-experts/issues/4#issuecomment-567124805

**Comment by davidmrau** [comment]
@sofiapvcp I fixed the issue by replacing Log_Softmax with a Softmax
